File: QF-Test/qftest-6.0.3/javascript/databinder.js
exports.__esModule = true;
var module$input = exports;
DataBinder = function() {
  _WrappedDataBinder = Java.type("de.qfs.apps.qftest.script.modules.DataBinder");
  var __wrappedInstance;
  function DataBinder() {
    __wrappedInstance = _WrappedDataBinder.instance();
  }
  DataBinder.prototype.bindList = parameterfy(function(rc, loopname, varname, values, separator, counter, intervals) {
    if (separator === undefined) {
      separator = null;
    }
    if (counter === undefined) {
      counter = null;
    }
    if (intervals === undefined) {
      intervals = null;
    }
    __wrappedInstance.bindList(rc, loopname, varname, values, separator, counter, intervals);
  });
  DataBinder.prototype.bindSets = parameterfy(function(rc, loopname, varnames, values, separator, counter, intervals) {
    if (separator === undefined) {
      separator = null;
    }
    if (counter === undefined) {
      counter = null;
    }
    if (intervals === undefined) {
      intervals = null;
    }
    __wrappedInstance.bindSets(rc, loopname, varnames, values, separator, counter, intervals);
  });
  DataBinder.prototype.bindDict = parameterfy(function(rc, loopname, dict, counter, intervals) {
    if (counter === undefined) {
      counter = null;
    }
    if (intervals === undefined) {
      intervals = null;
    }
    __wrappedInstance.bindDict(rc, loopname, dict, counter, intervals);
  });
  return DataBinder;
}();
var DataBinder_export = new DataBinder;
var $jscompDefaultExport = DataBinder_export;
module$input.DataBinder = DataBinder_export;
module$input["default"] = $jscompDefaultExport;

File: QF-Test/qftest-6.0.3/jython/Lib/databinder.py
from de.qfs.apps.qftest.script.modules import DataBinder as _WrappedDataBinder
#  redirected constants
#  redirected methods
__wrappedInstance = _WrappedDataBinder.instance()
def bindList(rc, loopname, varname, values, separator=None, counter=None, intervals=None):
    """
    Create and register a databinder that binds a list of values to a
    variable.
    
    @param varname
    The name of the variable to bind to.
    @param values
    The values to bind. Either a sequence type or a string to
    split.
    @param separator
    Separator character to split the values at in case they're a
    string. Default is whitespace.
    @param counter
    An optional variable name for the iteration counter.
    @param intervals
    Optional ranges of indices, separated by comma, e.g. "0,2-3".
    @throws TestException
    """
    __wrappedInstance.bindList(rc, loopname, varname, values, separator, counter, intervals)

def bindSets(rc, loopname, varnames, values, separator=None, counter=None, intervals=None):
    """
    Create and register a databinder that binds a list of value-set to a set
    of variables.
    
    @param loopname
    The name for which to register the databinder.
    @param varnames
    The names of the variables to bind to. Either a sequence type
    or a string to split.
    @param values
    The values to bind. Either a sequence of sequences - each
    inner sequence being one set of data to bind - or a string to
    split.
    @param separator
    Separator character to split the varnames and value-sets at in
    case they're a string. Default is space. Value-sets are
    separated by linebreaks.
    @param counter
    An optional variable name for the iteration counter.
    @param intervals
    Optional ranges of indices, separated by comma, e.g. "0,2-3".
    @throws TestException
    """
    __wrappedInstance.bindSets(rc, loopname, varnames, values, separator, counter, intervals)

def bindDict(rc, loopname, dict, counter=None, intervals=None):
    """
    Create and register a databinder that binds a dictionary. The keys of the
    dictionary are the names of the variables and the values are sequences of
    values to be bound.
    
    @param loopname
    The name for which to register the databinder.
    @param dict
    The dictionary to bind.
    @param counter
    An optional variable name for the iteration counter.
    @param intervals
    Optional ranges of indices, separated by comma, e.g. "0,2-3".
    @throws TestException
    """
    __wrappedInstance.bindDict(rc, loopname, dict, counter, intervals)

